### ISSUE TITLE
No service for type 'Microsoft.Framework.Runtime.IApplicationEnvironment'

### ISSUE THREAD

**Issue by javi830810-tsunami** [aspnet/Home#588]
I have this issue, after update to beta5, this happened to a working project in beta4.
More details:
1 - Tried on empty VM from scratch. No luck
2 - Output of dnvm list

```
 $ dnvm list
Active Version              Runtime Arch Location             Alias
------ -------              ------- ---- --------             -----
       1.0.0-beta4          mono         ~/.dnx/runtimes
  *    1.0.0-beta5-11749    mono         ~/.dnx/runtimes      default
```

3 -  The exception happens before reaching the entry point of my WebApp. I dont have control over what's happening.

Any help could be really appreciated. I've honestly tried as much as I could, even tried to downgrade to beta4 with no luck.

Below you can see the stack trace:

```
System.InvalidOperationException: No service for type 'Microsoft.Framework.Runtime.IApplicationEnvironment' has been registered.
  at Microsoft.Framework.DependencyInjection.ServiceProviderExtensions.GetRequiredService (IServiceProvider provider, System.Type serviceType) [0x00000] in <filename unknown>:0
  at Microsoft.Framework.DependencyInjection.ServiceProviderExtensions.GetRequiredService[IApplicationEnvironment] (IServiceProvider provider) [0x00000] in <filename unknown>:0
  at Microsoft.AspNet.Hosting.HostingEngine..ctor (IServiceProvider fallbackServices) [0x00000] in <filename unknown>:0
  at Microsoft.AspNet.Hosting.Program.Main (System.String[] args) [0x00000] in <filename unknown>:0
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <filename unknown>:0
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x00000] in <filename unknown>:0
  at Microsoft.Framework.Runtime.Common.EntryPointExecutor.Execute (System.Reflection.Assembly assembly, System.String[] args, IServiceProvider serviceProvider) [0x00000] in <filename unknown>:0
  at Microsoft.Framework.ApplicationHost.Program.ExecuteMain (Microsoft.Framework.Runtime.DefaultHost host, System.String applicationName, System.String[] args) [0x00000] in <filename unknown>:0
  at Microsoft.Framework.ApplicationHost.Program.Main (System.String[] args) [0x00000] in <filename unknown>:0
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x00000] in <filename unknown>:0
  at Microsoft.Framework.Runtime.Common.EntryPointExecutor.Execute (System.Reflection.Assembly assembly, System.String[] args, IServiceProvider serviceProvider) [0x00000] in <filename unknown>:0
  at dnx.host.Bootstrapper.RunAsync (System.Collections.Generic.List`1 args) [0x00000] in <filename unknown>:0
```

**Comment by davidfowl** [comment]
> I have this issue, after update to beta5, this happened to a working project in beta4.

What sdk version is in your global.json?

**Comment by javi830810-tsunami** [comment]
@davidfowl Thanks for so prompt response. 

*None* version: this is my global.json
{
    "projects": ["src"]
}

**Comment by davidfowl** [comment]
When you don't have a version pinned in `global.json`, visual studio will use the `default` alias (`dnvm list`). So if you did `dnvm upgrade -u`, you'll break your beta4 applications that had unpinned versions in `global.json`. To pin the version change your global.json to this:

```JSON
{
    "projects": [ "src" ],
    "sdk": {
        "version": "1.0.0-beta4"
    }
}
```

That will make sure VS launches 1.0.0-beta4 for this sample.

**Comment by javi830810-tsunami** [comment]
I'm really sorry, perhaps I wasnt clear enough, but this is happenning to me in both: OSX and Ubuntu 14-04 (the Ubuntu VM I set it up, today) the OSX machine have been my working machine for vNext for 2 months and it was working until I did dnvm upgrade -u.

I def made the change you requested although no luck, in OSX I dont think dnx . kestrel takes into account global.json. It's there a way I can specify the specific version of the DNX at the project level?
